lib/segment/src/common/operation_time_statistics.rs

```diff
@@ -87,9 +138,39 @@ impl std::ops::Add for OperationDurationStatistics {
                 other.max_duration_micros,
                 |a, b| a > b,
             ),
+            total_duration_micros: self.total_duration_micros + other.total_duration_micros,
             last_responded: std::cmp::max(self.last_responded, other.last_responded),
+            duration_micros_histogram: merge_buckets(
+                self.duration_micros_histogram,
+                other.duration_micros_histogram,
+            ),
+        }
+    }
+}
+
+/// Merges two vectors of buckets, summing the counts of the same boundaries.
+/// # Panics
+/// Panics when merging buckets of different configurations.  In practice, we should stick to a
+/// single configuration so this should not happen.
+fn merge_buckets(mut a: Vec<(f32, usize)>, b: Vec<(f32, usize)>) -> Vec<(f32, usize)> {
+    if a.is_empty() {
+        return b;
+    }
+    if b.is_empty() {
+        return a;
+    }
+    if a.len() != b.len() {
+        eprintln!("a = {a:?}, b = {b:?}");
+        panic!("Cannot merge buckets of different lengths");
+    }
```

Comment:
We should not `println` nor panic.

For logging we can use `log::error!()` (or whichever makes more sense).
We can to return an `OperationResult` here with a `ServiceError`.

I understand that this is very unlikely, but if we _can_ prevent runtime crashes by propagating errors, we should.

The same goes for the panic below.